code/socialDistribution/views.py

```diff
@@ -357,6 +358,23 @@ def posts(request, author_id):
     # In this case, app_name is socialDistribution
     return redirect('socialDistribution:home')
 
+# https://books.agiliq.com/projects/django-orm-cookbook/en/latest/copy.html - How to copy or clone an existing model object
+def sharePost(request, id):
+    """
+        Allows user to share a post.
+        The user that is sharing is the owner of the shared post
+        Public posts are shared to everyone
+        Friend posts are shared to friends
+    """
+    author = LocalAuthor.objects.get(user=request.user)
+    post = Post.objects.get(id=id)
+    post.pk = None # duplicate the post
+    post.author = author
+    post.pub_date = datetime.now()
+    post.save()
```

Comment:
I believe we'll also need to update the source field here. The ID of the post we are sharing will become the new source (since that's where we got it from)?

Also, now that the other PR is merged, can we update this PR to use InboxPost and dispatch the created post to the inbox's of other authors?

Reply:
I updated the function to use the new dispatch_post method. Is this the right way to dispatch now?

Reply:
Ya, that should be good 👍